operators/src/util/raster_stream_to_geotiff.rs

```diff
@@ -112,6 +112,10 @@ fn gdal_writer<T: Pixel + GdalType>(
     as_cog: bool,
 ) -> Result<()> {
     const INTERMEDIATE_FILE_SUFFIX: &str = "GEO-ENGINE-TMP";
+    const COG_BLOCK_SIZE: &str = "512";
+    const COMPRESSION_FORMAT: &str = "LZW";
+    const COMPRESSION_LEVEL: &str = "9"; // maximum compression
+    const COMPRESSION_NUM_THREADS: &str = "ALL_CPUS"; // full power for compression
```

Comment:
should this really be a default? it will use all threads. if we handle 100 compression tasks this could cause problems

Reply:
We could either go back to one thread or we use a config setting for this.
What is your favorite?

Reply:
config setting and a safe default like 8 i guess...

Reply:
9a953a1 

I added the config value.

Moreover, I refactored the functions that were too long either in their body or their signature to get rid of the Clippy warnings.